src/banking_stage.rs

```diff
@@ -112,31 +106,28 @@ impl BankingStage {
                                         break Some(BankingStageReturnType::ChannelDisconnected);
                                     }
                                     Error::BankError(BankError::RecordFailure) => {
-                                        break Some(BankingStageReturnType::RecordFailure);
+                                        warn!("Bank failed to record");
+                                        break Some(BankingStageReturnType::ChannelDisconnected);
+                                    }
+                                    Error::BankError(BankError::MaxHeightReached) => {
+                                        // Bank has reached its max tick height.  Exit quietly
+                                        // and wait for the PohRecorder to start leader rotation
+                                        break None;
                                     }
-                                    Error::PohRecorderError(PohRecorderError::MaxHeightReached) => {
-                                        if !thread_did_notify_rotation.load(Ordering::Relaxed) {
-                                            // Leader rotation should only happen if a max_tick_height was specified
-                                            let _ = thread_sender.send(
-                                                TpuReturnType::LeaderRotation(max_tick_height),
-                                            );
-                                            thread_did_notify_rotation
-                                                .store(true, Ordering::Relaxed);
-                                        }
-
-                                        //should get restarted from the channel receiver
-                                        break Some(BankingStageReturnType::LeaderRotation(
-                                            max_tick_height,
-                                        ));
+                                    _ => {
+                                        error!("solana-banking-stage-tx: unhandled error: {:?}", e)
                                     }
-                                    _ => error!("solana-banking-stage-tx {:?}", e),
                                 }
                             }
                             if thread_banking_exit.load(Ordering::Relaxed) {
                                 break None;
                             }
                         };
-                        thread_banking_exit.store(true, Ordering::Relaxed);
+
+                        // Signal exit only on "Some" error
+                        if return_result.is_some() {
```

Comment:
Here's where we avoid shutting down the rest of the Tpu if/when the Bank hits MaxHeightReached before the PohRecorder does.

Reply:
Yup okay makes sense.